### ISSUE TITLE
Python 3.7

### ISSUE THREAD

**Issue by joaoparma** [fnzr/minha-biblioteca-jigsaw#1]
Hello.

How can I run it for Python 3.7?
I tried to do it but stopped here...

virtualenv -p python3 venv

it says "The path python3 (from --python=python3) does not exist"

Please Help!

**Comment by fnzr** [comment]
Well I haven't used this in a while.
I pushed a commit that fixes the image download, so make sure to update.
The `python3` argument is the python executable. If your python is just python, write `virtualenv -p python venv`. I just tested with 3.7 and it seems to be fine, but let me know if you have any issues.

**Comment by joaoparma** [comment]
Thank you for your answer...I don't know anything about Python or any other code..just started this Tuesday.

Another question...Do I need to put python.exe on the jigsaw folder?

**Comment by fnzr** [comment]
I see. A couple steps back, then. Are you on Windows? Because this instruction (`. venv/bin/activate` ) is Linux-only. If you're on Windows, you have to use `.\venv\Scripts\activate.bat` instead.
Type `python --version` on the command prompt. If you see "Python 3.7.X", then you don't have to add python.exe to the project folder and you're all set.
If you're stuck in a specific step, tell me what it is or post the error message so I can give more directed instructions.

**Comment by joaoparma** [comment]
I'm almost there... after running python main.py :isbn this is what I see.

![capturar](https://user-images.githubusercontent.com/47673350/52899431-3e9e5300-31d1-11e9-9d6f-88d79a99fe93.PNG)

**Comment by fnzr** [comment]
Ok two things.
First, my bad. The book's ISBN is not prefixed by `:`. Just put the number in there. I'll fix the readme.
Second, it does seem this book format is not supported. I'll take a look a get back to you.

**Comment by joaoparma** [comment]
Thank you! It is downloading now!

Do I need to type this after downloading all pages? 

./[timestamp]/Text/[book_title].html

**Comment by fnzr** [comment]
Oh alright then, that's great.
No, after the download is done the script will write the resulting file name for you. Hopefully.
I don't really remember, so let me know if it all goes well, that would be cool.

**Comment by joaoparma** [comment]
Everything went well. Thank you once again!

**Comment by joaoparma** [comment]
Do you have something for this site ?https://bv4.digitalpages.com.br/#/

Will the program work the same way?

**Comment by fnzr** [comment]
Nope, this other site does things very differently, so it will definitely not work.

**Comment by joaoparma** [comment]
Hello!

I think I'm facing some problems here...I Can't download... Can you help me?

![capturar](https://user-images.githubusercontent.com/47673350/52916748-4a1d7700-32c2-11e9-9ea3-9daffd6f4da0.PNG)

**Comment by fnzr** [comment]
You probably forgot to activate the virtual env with `.\venv\Scripts\activate.bat`

**Comment by MasterCalcioXT** [comment]
![capturar](https://user-images.githubusercontent.com/47673350/52916923-04fa4480-32c4-11e9-8f4f-a81278e36515.PNG)

**Comment by fnzr** [comment]
Did you download the repository again, or deleted the venv folder? Because if you did, you also have to run `pip install -r requirements.txt` after activating the environment, because the modules are installed inside venv folder.

**Comment by MasterCalcioXT** [comment]
Just to know if I got it.. (Problem is solved, thanks)

Every time I start this process over (ex. another day) I need to copy the set cookie form response headers and paste ir to cookie.txt (not always as you said in the tutorial but the first is necessary), then run  `.\venv\Scripts\activate.bat` and then `python` main.py isbn

Am I right?

Thank you for all your support!

**Comment by fnzr** [comment]
yep, that's right. The cookies are your authentication (keep them secret!) and they expire after a while, so sometimes you have to get the updated version of them from the browser.

**Comment by MasterCalcioXT** [comment]
Hey man...

Another trouble :/

After downloading all pages this is what I see...
![capturar](https://user-images.githubusercontent.com/47673350/52919393-dccd0e80-32e0-11e9-9149-e1527f0adc03.PNG)

**Comment by fnzr** [comment]
Oh wow, it seems you're running out of memory when creating the PDF file. That's weird.
You will be able to read the book with an image browser, try searching in one of the folders the script creates in the default directory. It will contain all the pages of the book as images.
I can't think of a solution to this off the top of my head. I'll try to play around with it if I find the time, but don't expect a quick response.

**Comment by MasterCalcioXT** [comment]
Oh...ok, thanks.

Do you think this might help? (take a look at pdf.py)

https://github.com/faustobdls/UFERSA-digitalpages

**Comment by fnzr** [comment]
Nope, to build the PDF file, they use the exact same strategy as I do, which is this loop:
```
for page in list_pages:
        print(page)
        pdf.add_page()
        pdf.image(str(page), 0, 0)
```

They don't check MemoryError, and at a glance they don't implement anything that could prevent it. Strictly speaking, I don't even know is MemoryError means *OutOf*MemoryError, I'd have to check FDPF documentation if there's some special meaning for it, but it's unlikely.
What book you're trying to download? If I get the same error I might get a clue to solve the problem.

**Comment by MasterCalcioXT** [comment]
Here's what I've tried...

978-85-216-1935-2
9788522128020
9788580551280

All of them with this same MemoryError

**Comment by amlemos** [comment]
> Do you have something for this site ?https://bv4.digitalpages.com.br/#/
> 
> Will the program work the same way?

Abra o console de desenvolvimento, procure um link que termina com p_ número .jpg, use uma extensão para baixar em massa a quantidade de páginas do livro, monte-o com um pdfcreator, pronto
